File: back/src/Padaria.ProductService/Models/Produto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Padaria.ProductService.Models
{
    public class Produto
    {
        public Guid Id { get; set; }
        public string Nome { get; set; } = string.Empty;
        public string Descricao { get; set; } = string.Empty;
        public decimal Preco { get; set; }
        public int Estoque { get; set; }
        public string UrlImagem { get; set; } = string.Empty;
        public bool Disponivel { get; set; } = true;
        public Guid CategoriaId { get; set; }
        public Categoria Categoria { get; set; } = null!;
        public DateTime DataCriacao { get; set; }
        public DateTime? DataAtualizacao { get; set; }
    }
}

File: front/padaria-web/src/app/core/models/Produto.ts
export interface Produto {
  id: string;
  nome: string;
  descricao: string;
  preco: number;
  estoque: number;
  urlImagem: string;
  disponivel: boolean;
  categoriaId: string;
  nomeCategoria: string;
  criadoEm: string;
}